src/odemis/acq/fastem.py

```diff
@@ -783,34 +787,49 @@ def correct_beam_shift(self):
         good beam positions we calculate in what direction and how much to shift beams, such that they are always
         centered on the mppc detector.
         """
+        pixel_size = self._ccd.pixelSize.value
+        magnification = self._lens.magnification.value
+        sigma = self._ccd.pointSpreadFunctionSize.value
         # asap=False: wait until new image is acquired (don't read from buffer)
         ccd_image = self._ccd.data.get(asap=False)
-
-        # Find the location of the spots on the diagnostic camera.
-        spot_coordinates, *_ = FindGridSpots(ccd_image, (8, 8))
-
-        # Transform the spots from the diagnostic camera coordinate system to a right-handed coordinate
-        # system with the origin in the bottom left.
-        spot_coordinates[:, 1] = ccd_image.shape[1] - spot_coordinates[:, 1]  # [px]
+        tform, error = estimate_grid_orientation_from_img(ccd_image, (8, 8), SimilarityTransform, sigma,
+                                                          threshold_rel=0.5)
+        logging.debug(f"Found center of grid at {tform.translation}, error: {error}.")
 
         # Determine the shift of the spots, by subtracting the good multiprobe position from the average (center)
         # spot position.
-        good_mp_position = (self._ccd.getMetadata()[model.MD_FAV_POS_ACTIVE]["x"],
-                            self._ccd.getMetadata()[model.MD_FAV_POS_ACTIVE]["y"])
-        shift = numpy.mean(spot_coordinates, axis=0) - good_mp_position  # [px]
-
+        fav_pos_active = self._ccd.getMetadata()[model.MD_FAV_POS_ACTIVE]
+        # If i and j are not in the metadata, use x and y instead.
+        try:
+            i = fav_pos_active["i"]
+        except KeyError:
+            i = fav_pos_active["x"]
+        try:
+            j = fav_pos_active["j"]
+        except KeyError:
+            j = ccd_image.shape[1] - fav_pos_active["y"]
```

Comment:
Why wouldn't it be there, while x/y would be there? Old version of the calibration function? If so, please add this info in the comment, and indicate the x/y code can be dropped soon.